pkgs/applications/networking/cluster/k3s/default.nix

```diff
@@ -205,9 +205,18 @@ let
     installPhase = ''
       pushd go/src/${goPackagePath}
 
-      mkdir -p "$out/bin"
+      mkdir -p $out/bin $out/.deps
       install -m 0755 -T ./dist/artifacts/k3s${archSuffix} "$out/bin/k3s"
 
+      # leave a reference to everything that is packaged in the cli that it
+      # then unpacks and uses at runtime, otherwise we could be missing runtime
+      # dependencies
+      cat > $out/.deps/nix-runtime-deps-refs <<EOF
+        ${k3sPlugins}
+        ${k3sBuildStage1}
+        ${runc}
```

Comment:
Couldn't these simply go to `propagatedBuildInputs`?

Reply:
Yup, indeed that works!

I had it in my head that `propagatedBuildInputs` would somehow only work for a build dependency, not also sneaking in a runtime dependency, but as you rightly pointed out, it does what I want.

Updated with that change, thanks!